InterGalacticCurrencyConvertor/src/main/java/DataProcessor.java

```diff
@@ -0,0 +1,80 @@
+package main.java;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+
+import main.java.query.Query;
+import main.java.statement.Statement;
+
+/**
+ * Processes user input based on its types: Statement or Query.
+ */
+public class DataProcessor {
+	private static ArrayList<Query> mQueryObjList = new ArrayList<>();
```

Comment:
Rename Query to QueryProcessor and Statement to StatementProcessor. e.g.

interface QueryProcessor {
  DataProcessResult process(Query);
}

Try to encapsulate String[] into some Query/Statement class.

Reply:
updated